workflows/fastqrepair.nf

```diff
@@ -21,74 +21,77 @@ workflow FASTQREPAIR {
 
     take:
     ch_samplesheet // channel: samplesheet read in from --input
-
     main:
-
     ch_versions = Channel.empty()
-    ch_multiqc_files = Channel.empty()
+    ch_decoupled = Channel.empty()
 
-    //
-    // MODULE: Run FastQC
-    //
+    // Decouple paired-end reads
+    ch_decoupled = ch_samplesheet.flatMap { metaData, filePaths -> filePaths.collect { file -> [metaData, file] } }
+
+    // Recover fastq.gz and skip *.fastq or *.fq
+    GZRT (
+        ch_decoupled
+    )
+
+    // Make fastq compliant and wipe bad characters
+    SCATTER_WIPE_GATHER (
+        GZRT.out.fastq
+    )
+
+    // Run if PAIRED-END reads only!
+    SCATTER_WIPE_GATHER.out.fixed_fastq
+        .branch {
+            single_end: it[0].single_end == true
+            paired_end: it[0].single_end == false
+        }
+        .set { filtered_ch }
+
+    // Remove unpaired reads and reads shorter than 20 nt
```

Comment:
That seems a very strict limit, is there a raeson why the the `20nt` is not customisable? (But not a blocker! I'm just curious!)